src/core/mapping/mapping.cc

```diff
@@ -220,7 +220,7 @@ void StoreMapping::populate_layout_constraints(
 {
   StoreMapping mapping{};
   mapping.policy = InstanceMappingPolicy::default_policy(target, exact);
-  mapping.stores.push_back(store);
+  mapping.stores.emplace_back(store);
```

Comment:
I'm a bit uncomfortable with the change here. Store has changed from a lightweight, copyable handle to a move-only class.  The reference_wrapper stuff here has potential lifetime issues. I'm sure the current implementation is correct, but I don't love creating a vector of references that returns from a function.

Can these type objects just be statics? They are going to get reused quite a bit, yeah?  Would it be better to move from `unique_ptr<Type>` to `Type*` where Type is an object with static lifetime?

Reply:
Note that we don't want to pass around pointers in the distributed setting, so `Type*` needs to be deserialized somehow in the mapper. We can certainly make a singleton type object for each distinct type, but then we need a centralized registry of `Type*` objects for the deserializer (so it can recover a `Type*` from its serialized form) , which would then introduce synchronization between the control code registering types and every deserialization attempt. I don't actually want to introduce this. Though I don't like how it's set up, I feel it's reasonable because store mappings cannot exist without stores nor can they outlive them.